haiku-apps/bdhcalc/bdhcalc-1.1.recipe

```diff
@@ -27,6 +27,10 @@ BUILD_PREREQUIRES="
 	cmd:g++
 	"
 
+if [ "$buildArchitecture"x == "x86_64"x ] ; then
+PATCHES="bdhcalc-1.1.patchset"
```

Comment:
Only x86_gcc2 use gcc2. And I believe we don't need an trailing `x`

```diff
-if [ "$buildArchitecture"x == "x86_64"x ] ; then
+if [ "$buildArchitecture" != "x86_gcc2" ] ; then
```

Reply:
I'd strongly recommend using `$effectiveTargetArchitecture` which is the same as `$targetArchitecture` for primary architectures. On recipes which do not support secondary architectures, this makes no difference. But on the other recipes, using `$buildArchitecture` (to decide e.g. if a patch is needed or not) is generally not appropriate.
Therefore, to keep consistency across all recipes, it is better to use the variable which works on all cases, and this is `$effectiveTargetArchitecture`.

On secondary arch builds, e.g. x86 second arch on x86_gcc2, we have:
```
effectiveTargetArchitecture='x86'
buildArchitecture='x86_gcc2'
targetArchitecture='x86_gcc2'
```
BTW, this is the output of `grep rchitecture= wrapper-script` inside the `work-x86-$portVersion` of any package being built on x86_gcc2 for the x86 secondary arch.